src/components/Forms/PasswordEditForm/index.tsx

```diff
@@ -0,0 +1,226 @@
+
+interface PasswordEditFormProps extends React.HTMLProps<HTMLElement> {
+  isOpen: boolean;
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const PasswordEditForm = ({
+  setIsOpen,
+  children,
+}: PasswordEditFormProps) => {
+  const supabase = useSupabaseClient<Database>();
+  const session = useSession();
+  const [isBeeingSaved, setIsBeeingSaved] = useState(false);
+
+  const [formData, setFormData] = useState<ResetCredentialsData>({
+    password: '',
+    email: '',
+    repeatPassword: '',
+    oldPassword: '',
+  });
+  const [
+    notification,
+    setNotification,
+  ] = useState<UserNotificationObjectType | null>(null);
+  useEffect(() => {
+    if (!notification) {
+      return;
+    }
+
+    const timeout = setTimeout(() => {
+      setNotification(_ => null);
+    }, 5000);
+    return () => clearTimeout(timeout);
+  }, [notification]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsBeeingSaved(true);
+    const updatePassword = async () => {
+      const { passwordIsValid } = validatePassword(formData.password);
+      if (!passwordIsValid) {
+        setNotification({
+          message: 'Dein neues Passwort ist nicht sicher genug',
+          type: 'error',
+        });
+        return;
+      }
+      const {
+        data: verifyUserData,
+        error: verifyUserError,
+      } = await supabase.auth.signInWithPassword({
+        email: session?.user?.email ?? '',
+        password: formData.oldPassword,
+      });
+      if (verifyUserError) {
+        setNotification(prev =>
+          createUserNotification({
+            dispatchedFrom: 'PasswordResetForm.handleSubmit',
+            message: verifyUserError.message.includes(
+              'Invalid login credentials'
+            )
+              ? 'Dein altes Passwort stimmt nicht'
+              : verifyUserError.message,
+            type: 'error',
+          })
+        );
+        console.error('Error verifying user:', verifyUserError.message);
+        return;
+      }
+
+      if (!verifyUserData) {
+        setNotification(_ =>
+          createUserNotification({
+            message: 'Fehler beim Verifizieren des Benutzers',
+            type: 'error',
+          })
+        );
+        console.error('Error verifying user');
+        return;
+      }
+
+      if (formData.password !== formData.repeatPassword) {
+        setNotification(_ =>
+          createUserNotification({
+            dispatchedFrom:
+              'PasswordResetForm.handleSubmit if (formData.password !== formData.repeatPassword)',
+            message: 'Passwörter stimmen nicht überein',
+            type: 'error',
+          })
+        );
+        // FIXME: This is not working. Why?
```

Comment:
@dnsos Maybe you have an idea why my notification is null at this point. The setNotification function seems not to change the state.

Reply:
After quite some investigation I found the issue: see [this commit](https://github.com/technologiestiftung/giessdenkiez-de/pull/562/commits/b6c00c9eba45c975f527f5a39fe35ac1249dd9d3) for a demonstration and explanation.

This [PR](https://github.com/technologiestiftung/giessdenkiez-de/pull/562) would actually make the notification appear by moving the password equality check before the call for creating a new session. But I'm not satisfied with this workaround. We should solve this in a more robust way. One idea could be to verify the old/current password without creating  a new session. Do you think there is a way to do so?